processing/src/main/java/org/apache/druid/query/metadata/metadata/SegmentAnalysis.java

```diff
@@ -173,10 +181,6 @@ public int hashCode()
   @Override
   public int compareTo(SegmentAnalysis rhs)
   {
-    // Nulls first
-    if (rhs == null) {
```

Comment:
why get rid of null handling?

Reply:
It doesn't work inside `compareTo()`, because obviously not symmetric. If somebody needs null handling then should use Comparators.nullsFirst()/nullsLast()